File: domain/src/main/scala/com/seanshubin/project/generator/domain/LineEmittingNotifications.scala
package com.seanshubin.project.generator.domain

import com.seanshubin.project.generator.domain.GlobalConstants.devonMarshaller

class LineEmittingNotifications(emit: String => Unit) extends Notifications {
  override def effectiveSpecification(specification: Specification.Project): Unit = {
    val lines = devonMarshaller.valueToPretty(specification)
    lines.foreach(emit)
  }
}

File: domain/src/main/kotlin/com/seanshubin/project/generator/domain/LineEmittingNotifications.kt
package com.seanshubin.project.generator.domain

class LineEmittingNotifications(private val emit:(String)->Unit):Notifications {
    override fun lookupVersionEvent(uriString: String, dependency: GroupArtifactVersion) {
        emit("group:${dependency.group} artifact:${dependency.artifact} version:${dependency.version} uri:$uriString")
    }
}
